### PR TITLE
=htc #20602 avoid checking racy value if we don't really care

### PR DESCRIPTION
**Pull request by ktoso**
Stabilizes https://github.com/akka/akka/issues/20602 somewhat

### CODE REVIEW

**Comment by ktoso** [comment]
So as I understand it the value of wasCompletedFuture is somewhat racy, and we don't care about it except when checking for `close = false` so I'm here avoiding to check it...

If you can think of a better solution, or dropping that check maybe (?) let me know @sirthias @jrudolph @2beaucoup

**Comment by drewhk** [comment]
The change itself looks OK, but I am missing the larger context. Can this be expressed differently?

**Comment by jrudolph** [comment]
LGTM

File: akka-http-core/src/test/scala/akka/http/impl/engine/rendering/ResponseRendererSpec.scala [hunk 1]
```diff
@@ -582,10 +582,13 @@ class ResponseRendererSpec extends FreeSpec with Matchers with BeforeAndAfterAll
...
 
     def renderTo(expected: String): Matcher[HttpResponse] =
-      renderTo(expected, close = false) compose (ResponseRenderingContext(_))
+      renderToImpl(expected, checkClose = None) compose (ResponseRenderingContext(_))
 
     def renderTo(expected: String, close: Boolean): Matcher[ResponseRenderingContext] =
-      equal(expected.stripMarginWithNewline("\r\n") -> close).matcher[(String, Boolean)] compose { ctx ⇒
+      renderToImpl(expected, checkClose = Some(close))
+
+    def renderToImpl(expected: String, checkClose: Option[Boolean]): Matcher[ResponseRenderingContext] =
```
Comment (2beaucoup) [inline akka-http-core/src/test/scala/akka/http/impl/engine/rendering/ResponseRendererSpec.scala]: You can get rid of the first `renderTo` if you just use a default param for `checkClose` here and rename back to `renderTo`.

Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/impl/engine/rendering/ResponseRendererSpec.scala]: I don't think I mind the manual delegation here, it's just a spec :) Thanks for the comment!

**Comment by ktoso** [comment]
Thanks for having a look guys, I'll merge to unblock our PR queue a bit

**Pull request merged**